Modules/Filtering/ImageFilterBase/test/itkCastImageFilterTest.cxx

```diff
@@ -87,20 +116,39 @@ bool TestCastFromTo()
   ot.GoToBegin();
   while ( !it.IsAtEnd() )
     {
-    TInputPixelType  inValue  = it.Value();
-    TOutputPixelType outValue = ot.Value();
-    auto expectedValue = static_cast< TOutputPixelType >( inValue );
+    const TInputPixelType  inValue  = it.Value();
+    const TOutputPixelType outValue = ot.Value();
+
+    const auto expectedValue = static_cast< TOutputPixelType >( inValue );
+
```

Comment:
Is this the check that fixes the issue? Should a comment be added, that this statement skips comparisons for "cast"'s with undefined behavior.

Reply:
Separated changes based on request.